File: src/main/java/com/indiepost/controller/api/InitialDataController.java
package com.indiepost.controller.api;

import com.indiepost.dto.InitialData;
import com.indiepost.service.InitialDataService;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import javax.inject.Inject;

/**
 * Created by jake on 17. 1. 22.
 */
@RestController
@RequestMapping("/api/init")
public class InitialDataController {

    private final InitialDataService initialDataService;

    @Inject
    public InitialDataController(InitialDataService initialDataService) {
        this.initialDataService = initialDataService;
    }

    @GetMapping
    public InitialData getInitialData() {
        return initialDataService.getInitialData(true);
    }
}

File: src/main/kotlin/com/indiepost/controller/InitialDataController.kt
package com.indiepost.controller

import com.indiepost.dto.InitialData
import com.indiepost.service.InitialDataService
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

import javax.inject.Inject

/**
 * Created by jake on 17. 1. 22.
 */
@RestController
@RequestMapping("/init")
class InitialDataController @Inject
constructor(private val initialDataService: InitialDataService) {

    @GetMapping
    fun getInitialData(): InitialData = initialDataService.getInitialData(withLatestPosts = true)
}
